src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAssignableTypeForForeachVariableTests.cs

```diff
@@ -0,0 +1,466 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Runtime.CSharpUseAssignableTypeForForeachVariable,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseAssignableTypeForForeachVariableTests
+    {
+        [Fact]
+        public async Task SameType()
+        {
+            var test = @"
+using System.Collections.Generic;
+
+namespace ConsoleApplication1
+{
+    class Program
+    {   
+        void Main()
+        {
+            var x = new List<string>();
+            foreach (string item in x)
+            {
+            }
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(test);
+        }
+
+        [Fact]
+        public async Task CastBaseToChild()
+        {
+            var test = @"
+using System.Collections.Generic;
+
+namespace ConsoleApplication1
+{
+    class Program
+    {   
+        void Main()
+        {
+            var x = new List<A>();
+            foreach (B item in x)
+            {
+            }
+        }
+    }
+
+    class A { }
+    class B : A { }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(test, GetCSharpResultAt(11, 13).WithArguments("A", "B"));
```

Comment:
Consider using VerifyCodeFixAsync and pass both the source and fixedSource the same value.

This will ensure that:

- *If* a code fix is added and it's applicable to this case, the tests will fail and be updated with the codefix behavior.
- *If* a code fix is added but it's not applicable to this test, the test will only fail **if** the codefix provides an action that does nothing, and will pass if no action is offered.

Reply:
Thanks for suggestion. Replaced `VerifyAnalyzerAsync` with `VerifyCodeFixAsync`